llvm/lib/Support/Unix/Watchdog.inc

```diff
@@ -19,13 +19,13 @@
 namespace llvm {
 namespace sys {
 Watchdog::Watchdog(unsigned int seconds) {
-#ifdef HAVE_UNISTD_H
+#if HAVE_UNISTD_H && !defined(__wasi__)
```

Comment:
Looks like this could again be switched to a WASI/Watchdog.inc, with stub impls.